nano/test_common/testutil.hpp

```diff
@@ -42,6 +42,16 @@
 		ASSERT_NO_ERROR (system.poll ()); \
 	}
 
+/*
+ * Waits specified number of time while keeping system running.
+ * Useful for asserting conditions that should still hold after some delay of time
+ */
+#define WAIT(time)              \
```

Comment:
this code is a duplicate. We already have system::delay_ms.
However, this implementation looks better. It should replace the implementation in delay_ms.

Reply:
Yup it's more ergonomic to use, just write `WAIT(1s)` instead of `system.delay_ms (std::chrono::milliseconds (1000));`